packages/property-provider/src/ProviderError.ts

```diff
@@ -8,6 +8,7 @@
  * ensures the chain will stop if an entirely unexpected error is encountered.
  */
 export class ProviderError extends Error {
```

Comment:
A ProviderError is essentially error created by Provider, and shouldn't be named CredentialsError.

Should we create a CredentialsError which extends ProviderError?